language/vm/vm-runtime/src/code_cache/module_cache.rs

```diff
@@ -551,6 +551,6 @@ where
     }
 
     fn reclaim_cached_module(&self, _v: Vec<LoadedModule>) {
-        panic!("reclaim_cached_module should never be called on TransactionModuleCache");
+        unreachable!("reclaim_cached_module should never be called on TransactionModuleCache");
```

Comment:
unreachable has the same runtime effect here, but MIRAI treats it as an explicit assumption made by the programmer, rather than as something that has to be proved as impossible to happen.